scipy/sparse/sparsetools/csr.h

```diff
@@ -1,12 +1,15 @@
 #ifndef __CSR_H__
 #define __CSR_H__
 
+#include <sstream>
 #include <set>
 #include <vector>
 #include <algorithm>
 #include <functional>
 #include <numeric>
 
+#include <numpy/npy_common.h>
```

Comment:
Not immediately clear to me without some digging whether it matters to avoid pulling in Python/NumPy C APIs here. It's only for `NPY_UNLIKELY` though, which is a single macro that can be replicated here instead if needed:
```C
#ifdef HAVE___BUILTIN_EXPECT
#define NPY_LIKELY(x) __builtin_expect(!!(x), 1)
#define NPY_UNLIKELY(x) __builtin_expect(!!(x), 0)
#else
#define NPY_LIKELY(x) (x)    
#define NPY_UNLIKELY(x) (x)
#endif
```